src/model/platform-setup/setup-mac.ts

```diff
@@ -110,17 +110,23 @@ class SetupMac {
     }
 
     const unityChangeset = await getUnityChangeset(buildParameters.editorVersion);
-    const moduleArgument = SetupMac.getModuleParametersForTargetPlatform(buildParameters.targetPlatform);
+    const moduleArguments = SetupMac.getModuleParametersForTargetPlatform(buildParameters.targetPlatform);
 
-    const command = `${this.unityHubExecPath} -- --headless install \
-                                          --version ${buildParameters.editorVersion} \
-                                          --changeset ${unityChangeset.changeset} \
-                                          ${moduleArgument} \
-                                          --childModules `;
+    const execArguments: string[] = [
+      '--',
+      '--headless',
+      'install',
+      ...['--version', buildParameters.editorVersion],
+      ...['--changeset', unityChangeset.changeset],
+      ...moduleArguments,
+      '--childModules',
+    ];
+
+    const escapedExecPath = this.unityHubExecPath.replace(/ /g, '\\ ');
 
     // Ignoring return code because the log seems to overflow the internal buffer which triggers
     // a false error
-    const errorCode = await exec(command, undefined, { silent, ignoreReturnCode: true });
+    const errorCode = await exec(escapedExecPath, execArguments, { silent, ignoreReturnCode: true });
```

Comment:
A bit of bikeshedding, but perhaps we could keep the argument names as `exec(command, arguments, {...`

Reply:
I first tried `arguments`, but current linter config complained me not to use that name.